errno/errname.go

```diff
@@ -826,6 +826,7 @@ var MySQLErrName = map[uint16]string{
 	ErrGeneratedColumnNonPrior:                               "Generated column can refer only to generated columns defined prior to it.",
 	ErrDependentByGeneratedColumn:                            "Column '%s' has a generated column dependency.",
 	ErrGeneratedColumnRefAutoInc:                             "Generated column '%s' cannot refer to auto-increment column.",
+	ErrExpressionIndexCanNotRefer:                            "Expression index '%s' cannot refer to an auto-increment column.",
```

Comment:
In MySQL 8.0.21, the error message is `Functional index 'idx1' cannot refer to an auto-increment column.`, and the error code is `3754` instead of `3110`:

```sql
MySQL(root@127.0.0.1:test) > CREATE TABLE t1 (t1_id INT NOT NULL AUTO_INCREMENT PRIMARY KEY);
Query OK, 0 rows affected (0.01 sec)

MySQL(root@127.0.0.1:test) > CREATE INDEX idx1 ON t1 ((t1_id + t1_id));
ERROR 3754 (HY000): Functional index 'idx1' cannot refer to an auto-increment column.
MySQL(root@127.0.0.1:test) > select version();
+-----------+
| version() |
+-----------+
| 8.0.21    |
+-----------+
1 row in set (0.00 sec)
```